invokeai/app/invocations/controlnet_image_processors.py

```diff
@@ -96,23 +98,29 @@
 class ControlField(BaseModel):
     image: ImageField = Field(default=None, description="The control image")
     control_model: Optional[str] = Field(default=None, description="The ControlNet model to use")
-    control_weight: Optional[float] = Field(default=1, description="The weight given to the ControlNet")
+    # control_weight: Optional[float] = Field(default=1, description="weight given to controlnet")
+    control_weight: Union[float, List[float]] = Field(default=1, description="The weight given to the ControlNet")
```

Comment:
All polymorphic fields that have values with constraints should have validators: https://docs.pydantic.dev/latest/usage/validators/

So for example, if `control_weight` should be >= 0 and <=1, the validator would:
- check if the value is a list
- if so, ensure all values in the list are valid
- if not a list, ensure the value is a valid value

You'd omit the constraints within the `Field()` definition if you use a custom validator.

This needs to be done for `cfg_scale` as well.

On a related note, please ensure that all fields use as strict of constraints as possible for all node schemas

Reply:
Added validators for control_weight and cfg_scale